src/test/scala/test_util/BASILTest.scala

```diff
@@ -0,0 +1,136 @@
+package test_util
+
+import org.scalatest.funsuite.AnyFunSuite
+import ir.{Block, Procedure, Program}
+import util.{BASILConfig, BASILResult, BoogieGeneratorConfig, ILLoadingConfig, Logger, RunUtils, StaticAnalysisConfig}
+
+import scala.sys.process.*
+import scala.io.Source
+import java.io.{BufferedWriter, File, FileWriter}
+
+case class TestConfig(boogieFlags: Seq[String] = Seq("/timeLimit:10", "/useArrayAxioms"),
+                      staticAnalysisConfig: Option[StaticAnalysisConfig] = None,
+                      useBAPFrontend: Boolean,
+                      expectVerify: Boolean,
+                      checkExpected: Boolean = false,
+                      logResults: Boolean = false
+                     )
+
+trait BASILTest extends AnyFunSuite {
```

Comment:
This unnecessarily restricts you to using a particular suite type in order to use these functions. These can just be defined as functions and imported rather than inherited.